src/components/html-text-input.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @format
+ * @flow
+ */
+
+import React from 'react';
+import { Platform, TextInput, KeyboardAvoidingView } from 'react-native';
+import styles from './html-text-input.scss';
+
+// Gutenberg imports
+import { serialize } from '@wordpress/blocks';
+
+type PropsType = {
+	blocks: Array<Object>,
+	parseBlocksAction: string => mixed,
+};
+
+type StateType = {
+	html: string,
+};
+
+export default class HTMLInputView extends React.Component<PropsType, StateType> {
+	state = {
+		html: '',
+	}
+
+	componentDidMount() {
+		const html = this.serializeBlocksToHtml();
+		this.setState( { html } );
+	}
+
+	componentWillUnmount() {
+		//TODO: Blocking main thread
+		this.props.parseBlocksAction( this.state.html );
+	}
+
+	shouldComponentUpdate() {
+		const isIOS = Platform.OS === 'ios';
+		if ( isIOS ) {
+			// iOS TextInput gets jumpy if it's updated on every key stroke.
+			return this.state.html === '';
```

Comment:
Not a blocker for the PR but, I wonder if it's intentional that we're checking the _current_ state for empty html instead of checking the _next_ state.

In case it was not intentional, should we check the _next_ State? `shouldComponentUpdate` accepts the next Props and State as parameters we can check. See https://reactjs.org/docs/react-component.html#shouldcomponentupdate.

Reply:
Yes! It is intentional. The first render of the component will always be empty. And since on iOS we were blocking updates, it was never showing content. 
So the logic is: If the previous state was empty, we let update the component to show the next state.

Since I had to explain it here, probably I should explain it better in code too. 😆